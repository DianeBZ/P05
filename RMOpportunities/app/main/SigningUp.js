import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  TouchableHighlight,
  BackAndroid,
  Dimensions,
} from 'react-native';
import Translation from './Translation';
import SigningUpTwoSteps from './SigningUpTwoSteps';

var width = Dimensions.get('window').width;
var height = Dimensions.get('window').height;
// this page is used to let user to sign up by his Linkedin account or directly
export default class SigningUp extends Component {
  // this function will be called when pressing the button sign up with Linkedin
  _onPressBouton(){
    console.log('press Linkedin');
  }

  render() {
    BackAndroid.addEventListener('backToIndex', this.onBackAndroid);
    return (
      <View style={styles.container1}>
        <View style={{height: height * 0.08}}/>
        <Text style={styles.titre}>
            {Translation[lang].inscript}
        </Text>
        <View style={styles.subContainer}>
          <View style={styles.sContainerPre}>
            <Text style={styles.presentation}>
                {Translation[lang].text_ins}
            </Text>
          </View>
          <View style={styles.sContainerBoutons}>
            <TouchableHighlight style={[styles.button, {width: width *0.85, backgroundColor:'rgb(0,160,220)'}]} onPress={this._onPressBouton} underlayColor="rgb(0,140,201)">
              <Text style={styles.buttonText}>
                  {Translation[lang].ins_linkedin}
              </Text>
            </TouchableHighlight>
            <Text style={{color:'grey', marginTop:20}}>
                {Translation[lang].ou}
            </Text>
            <TouchableHighlight style={[styles.button, {width: width *0.85, backgroundColor:'#A4D04A'}]} onPress={this.onPressInscription} underlayColor="rgb(138,183,46)">
              <Text style={styles.buttonText}>
                  {Translation[lang].ins2etapes}
              </Text>
            </TouchableHighlight>
          </View>
        </View>
      </View>
    );
  }

  onBackAndroid = () => {
	const { navigator } = this.props;
    if (navigator && navigator.getCurrentRoutes().length > 1) {
      BackAndroid.removeEventListener('backToIndex', this.onBackAndroid);
      navigator.pop();
      return true;
    } else {
      return false;
    }
  };
// this function will be called when pressing 'Sign Up in 2 Steps' and the current page will be 'SigningUpTwoSteps'.

  onPressInscription = () => {
		BackAndroid.removeEventListener('Exit',this.onBackAndroid);
		const { navigator } = this.props;
		if (navigator) {
			navigator.push({
				name: 'SigningUpTwoSteps',
				component: SigningUpTwoSteps,
			})
		}
	};
}

const styles = StyleSheet.create({
  container1: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'center',
    backgroundColor: '#F2F2F2',
  },
  subContainer:{
    backgroundColor:'#FFFFFF',
    flex: 1,
  },
  sContainerPre:{
    backgroundColor:'#A4D04A',
    height: height*0.16,
    justifyContent: 'center',
  },
  sContainerBoutons:{
    marginTop: height*0.06,
    justifyContent: 'flex-start',
    alignItems: 'center',
  },
   titre:{
       fontSize: 32,
       textAlign: 'center',
       color: '#000000',
       marginTop: 15,
       marginBottom: 15
   },
  presentation: {
    textAlign: 'center',
    color:"black",
    fontSize: 20,
  },
  button:{
        height: height*0.05,
        marginTop: 20,
        borderRadius:5,
        justifyContent: 'center',
    },
    buttonText:{
        color: "white",
        textAlign: "center",
        fontSize: 20,
    }
});
