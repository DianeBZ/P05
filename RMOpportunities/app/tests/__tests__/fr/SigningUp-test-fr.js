// __tests__/SigningUp-test-fr.js
import 'react-native';
import React from 'react';

global.lang = 'fr';
import SigningUp from '../../../main/SigningUp';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <SigningUp />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
