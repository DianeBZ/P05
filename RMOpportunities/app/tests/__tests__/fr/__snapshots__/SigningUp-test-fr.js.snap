// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F2F2F2",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      Object {
        "height": 106.72,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#000000",
        "fontSize": 32,
        "marginBottom": 15,
        "marginTop": 15,
        "textAlign": "center",
      }
    }
  >
    Inscrivez-vous
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#A4D04A",
          "height": 213.44,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        L'inscription de votre société sur RMOpportunities est simple, rapide et gratuite
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "flex-start",
          "marginTop": 80.03999999999999,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "borderRadius": 5,
                "height": 66.7,
                "justifyContent": "center",
                "marginTop": 20,
              },
              Object {
                "backgroundColor": "rgb(0,160,220)",
                "width": 637.5,
              },
            ],
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          S'inscrire avec Linkedin
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "grey",
            "marginTop": 20,
          }
        }
      >
        OU
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "borderRadius": 5,
                "height": 66.7,
                "justifyContent": "center",
                "marginTop": 20,
              },
              Object {
                "backgroundColor": "#A4D04A",
                "width": 637.5,
              },
            ],
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          Démarrer l'inscription en 3 étapes
        </Text>
      </View>
    </View>
  </View>
</View>
`;
